server/src/main/java/org/elasticsearch/index/mapper/ParsedDocument.java

```diff
@@ -160,7 +161,7 @@ public void addDynamicMappingsUpdate(Mapping update) {
         if (dynamicMappingsUpdate == null) {
             dynamicMappingsUpdate = update;
         } else {
-            dynamicMappingsUpdate = dynamicMappingsUpdate.merge(update, MergeReason.MAPPING_UPDATE);
+            dynamicMappingsUpdate = dynamicMappingsUpdate.merge(update, MergeReason.MAPPING_UPDATE, NewFieldsBudget.unlimited());
```

Comment:
No, we'll apply the field limit in `MapperService#merge`, depending on whether the merge reason resembles an auto-update (dynamic mappings). See also https://github.com/elastic/elasticsearch/pull/96235/files#diff-32c1e36544e0c0ef775f9e26576ef7df0504cc05466e513ba7b143947c234784R590.

The only instance when `addDynamicMappingsUpdate` is called is during `prepareIndex` to ensure that there's an empty mapping:

https://github.com/elastic/elasticsearch/blob/78c6a7c4bb177d9e63f7c947a09e75cda6a616de/server/src/main/java/org/elasticsearch/index/shard/IndexShard.java#L1008-L1019

That's why this will remain to use no limit.

Ideally, I'd like to remove the method tbh or don't merge or override the mapping within `addDynamicMappingsUpdate`.